indexing-service/src/main/java/org/apache/druid/indexing/common/task/InputSourceProcessor.java

```diff
@@ -58,7 +61,7 @@ public class InputSourceProcessor
    *
    * @return {@link SegmentsAndCommitMetadata} for the pushed segments.
    */
-  public static SegmentsAndCommitMetadata process(
+  public static Pair<SegmentsAndCommitMetadata, SegmentSchemaMapping> process(
```

Comment:
**For follow-up PR**
We should include `SegmentSchemaMapping` inside the `SegmentsAndCommitMetadata` itself.